packages/core/src/html/UlNodeProcessor.ts

```diff
@@ -0,0 +1,352 @@
+  const {
+    isEmoji,
+    isImage,
+    unicodeEmoji,
+    iconClass,
+    iconName,
+    isOcticon,
+    octiconColor,
+    octiconClass,
+  } = classifyIcon(iIcon);
+  const icon = iconClass || iIcon;
+
+  let child: NodeOrText;
+  const defaultSize = `width: ${size.width}; height: ${size.height}; 
+          margin-right:5px;text-align:center;display:flex;align-items:center;`;
+
+  if (isEmoji) {
+    child = {
+      type: 'tag',
+      name: 'span',
+      attribs: {
+        ...className && { class: className },
+        'aria-hidden': 'true',
+        style: `${defaultSize}${size.fontSize ? `font-size:${size.fontSize};
+        margin-right: 5px;` : 'margin-right: 5px;'}`,
+      },
+      children: [
+        {
+          type: 'text',
+          data: unicodeEmoji,
+        },
+      ],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+  } else if (isImage) {
+    const altText = icon.split('/').pop()?.split('.')[0] || '';
+    const divNode: NodeOrText = {
+      type: 'tag',
+      name: 'div',
+      attribs: {
+        style: 'padding: 0.2rem;',
+      },
+      children: [],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+
+    const imgNode: NodeOrText = {
+      type: 'tag',
+      name: 'img',
+      attribs: {
+        ...className && { class: className },
+        src: icon,
+        alt: altText,
+        style: defaultSize,
+      },
+      children: [],
+      next: undefined,
+      prev: undefined,
+      parent: divNode,
+    };
+    divNode.children?.push(imgNode);
+    child = divNode;
+  } else if (isOcticon) {
+    const octiconIcon = getOcticonIcon(iconName!);
+    let svgStr = octiconIcon.toSVG({
+      class: octiconClass ? `${iconClass} ${octiconClass} ${className}` : `${iconClass} ${className}`,
+    });
+
+    if (octiconColor) {
+      svgStr = svgStr.replace('<path', `<path fill="${octiconColor}"`);
+    }
+    child = {
+      type: 'tag',
+      name: 'span',
+      attribs: {
+        'aria-hidden': 'true',
+        style: `${defaultSize}margin-right:5px;${size.fontSize ? `font-size:${size.fontSize};` : ''}`,
+      },
+      children: [
+        {
+          type: 'text',
+          data: svgStr,
+        },
+      ],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+  } else {
+    child = {
+      type: 'tag',
+      name: 'span',
+      attribs: {
+        class: className ? `${icon} ${className}` : icon,
+        'aria-hidden': 'true',
+        style: `${defaultSize}margin-right:5px;${size.fontSize ? `font-size:${size.fontSize};` : ''}`,
+      },
+      children: iconName ? [
+        {
+          type: 'text',
+          data: iconName,
+        },
+      ] : [],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+  }
+  return child;
+}
+
+function getSize(widthPar: string, heightPar: string) {
+  const width = widthPar || heightPar || '25px';
+  const height = heightPar || widthPar || '25px';
```

Comment:
For images/media,

Not sure if this has been updated yet - placing defaults here would also mean users would have to manually calculate image aspect ratios themselves contrary to to what we discussed in the [earlier](https://github.com/MarkBind/markbind/pull/2316#discussion_r1253171710) thread.

As an author, I should be able to specify only `width` _or_ `height` and let the other dimension be defined automatically according to the media size (similar to `<img>` tags).

If neither were specified, only then should we add a default `width` (or `height`).